### PR TITLE
Upgrade guzzle library to newest version.

### PR DESCRIPTION
**Pull request by dalailomo**
- modified implementation and tests using guzzle
- modified README file

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "require": {
         "php": ">=5.4",
```
Comment (gavinlove) [inline composer.json]: You need to change the PHP requirement as it does not work with PHP 5.4

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
+.idea/
```
Comment (rainerkraftmb) [inline .gitignore]: This should not be in here - if you want to ignore IDE files, put them into your global gitignore file

File: src/Service/WhenIWorkApi.php [hunk 3]
```diff
@@ -171,14 +171,12 @@ private function get($entryPoint)
...
-                ->send();
-
-            return $response->json();
+                self::WHEN_I_WORK_ENDPOINT . $entryPoint,
+                array('headers' => array('W-Token' => $this->apiToken))
+            );
 
-        } catch (ClientErrorResponseException $e) {
+            return \GuzzleHttp\json_decode($response->getBody()->getContents(), true);
+        } catch (BadResponseException $e) {
```
Comment (rainerkraftmb) [inline src/Service/WhenIWorkApi.php]: I think here we should also catch `RequestException`s

File: src/Service/WhenIWorkApi.php [hunk 4]
```diff
@@ -194,29 +192,23 @@ private function setUp()
...
-                $headers,
-                json_encode($params)
+                $options
             );
-            $response = $request->send();
-            $response = $response->json();
 
-        } catch (ClientErrorResponseException $e) {
+            $decodedResponse = \GuzzleHttp\json_decode($response->getBody()->getContents(), true);
+        } catch (BadResponseException $e) {
```
Comment (rainerkraftmb) [inline src/Service/WhenIWorkApi.php]: See similar comment above

**Pull request merged**